source/common/router/config_impl.cc

```diff
@@ -402,6 +406,13 @@ std::string RouteEntryImplBase::newPath(const Http::HeaderMap& headers) const {
   } else {
     ASSERT(headers.Path());
     final_path = headers.Path()->value().c_str();
+    if (strip_query_) {
+      path = headers.Path()->value().c_str();
+      size_t path_end = path.find("?");
+      if (path_end != std::string::npos) {
+        final_path = (path.erase(path_end)).c_str();
+      }
+    }
```

Comment:
If you made final_path an absl::string_view, I believe you could remove the need for the `path` variable and avoid a string copy.

Something like:
```
final_path = absl::string_view(header.Path()->value());
if (strip_query_) {
  size_t path_end = final_path.find('?')
  if (path_end != abs::string_view::npos) {
    final_path = final_path.substr(0, path_end);
  }
}
```

Reply:
Sure. Will do it.